### PR TITLE
feat: Add datadog logs in notification service

### PR DESCRIPTION
**Pull request by johnranj**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Add logging for notification.

### Solutions

Added datadog logs for notification service 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire Notification Service Extension/LegacyNotificationService.swift [hunk 1]
```diff
@@ -257,3 +258,24 @@ extension UNNotificationContent {
   }
 
 }
+
+extension DatadogWrapper: WireLogger {
+    
+    public func info(_ message: String) {
+        print("loging datadog...")
```
Comment (johnxnguyen) [inline Wire Notification Service Extension/LegacyNotificationService.swift:265]: Please remove this print statement.

File: Wire Notification Service Extension/LegacyNotificationService.swift [hunk 2]
```diff
@@ -257,3 +258,24 @@ extension UNNotificationContent {
...
+        self.log(level: .info, message: message)
```
Comment (johnxnguyen) [inline Wire Notification Service Extension/LegacyNotificationService.swift:266]: `self` is not needed, please remove.

File: Wire Notification Service Extension/LegacyNotificationService.swift [hunk 3]
```diff
@@ -257,3 +258,24 @@ extension UNNotificationContent {
...
+    public func debug(_ message: String) {
+        self.log(level: .debug, message: message)
+    }
+
+    public func warn(_ message: String) {
+        self.log(level: .warn, message: message)
+    }
+
+    public func error(_ message: String, _ error: Error) {
+        self.log(level: .critical, message: message)
```
Comment (johnxnguyen) [inline Wire Notification Service Extension/LegacyNotificationService.swift:278]: Is there an `error` level we could use?

File: Wire Notification Service Extension/LegacyNotificationService.swift (hunk 2 above)
Comment (johnranj) [reply Wire Notification Service Extension/LegacyNotificationService.swift]: its resolved now

**Review by agisilaos — APPROVED** [review]

**Pull request closed without merging**